### PR TITLE
Use transform to convert UserCode to SubmitUsercode in EnclaveService.setup_enclave_for_code_execution

### PR DESCRIPTION
**Pull request by BrendanSchell**
## Description
Adds transform for conversion from UserCode to SubmitUserCode for use in enclave service. 

## Affected Dependencies
List any dependencies that are required for this change.

## How has this been tested?
- Describe the tests that you ran to verify your changes.
- Provide instructions so we can reproduce.
- List any relevant details for your test configuration.

## Checklist
- [ ] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [ ] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [ ] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [ ] My changes are covered by tests

### CODE REVIEW

File: packages/syft/src/syft/service/code/user_code.py [hunk 1]
```diff
@@ -1527,6 +1527,42 @@ def submit_user_code_to_user_code() -> list[Callable]:
...
+            {v: context.output.get(k) for k, v in remap_attributes.items()}
+        )
+
+    context.output = new_context_output
+    return context
+
+
+@transform(UserCode, SubmitUserCode)
+def user_code_to_submit_user_code() -> list[Callable]:
+    return [copy_user_code_attributes_for_submit_user_code]
```
Comment (itstauq) [inline packages/syft/src/syft/service/code/user_code.py]: There are a few useful utility methods for common transform operations in `syft.types.transforms` that we can use here instead:
- `drop`
- `rename`
- `keep`

**Review by itstauq — CHANGES_REQUESTED** [review]

File: packages/syft/src/syft/service/code/user_code.py [hunk 2]
```diff
@@ -1527,6 +1529,31 @@ def submit_user_code_to_user_code() -> list[Callable]:
...
+                "func_name",
+                "signature",
+                "input_policy_type",
+                "input_policy_init_kwargs",
+                "output_policy_type",
+                "output_policy_init_kwargs",
+                "deployment_policy_type",
+                "deployment_policy_init_kwargs",
+                "input_kwargs",
+                "enclave_metadata",
```
Comment (itstauq) [inline packages/syft/src/syft/service/code/user_code.py]: This was part of the old Enclave implementation and is now redundant. We can remove it.

**Review by itstauq — APPROVED** [review]
LGTM 🫡

File: packages/syft/src/syft/service/code/user_code.py (hunk 2 above)
Comment (BrendanSchell) [reply packages/syft/src/syft/service/code/user_code.py]: ah ok, this is specifically `enclave_metadata` that can be removed?

Comment (itstauq) [reply packages/syft/src/syft/service/code/user_code.py]: Yep 💯

**Comment by rasswanth-s** [comment]
Great work @BrendanSchell

**Pull request merged**